Sources/Vortex/Settings/Settings.swift

```diff
@@ -0,0 +1,314 @@
+        public var attractionStrength: Double = .zero
+
+        /// How fast movement speed should be slowed down.
+        /// Defaults to 0
+        public var dampingFactor: Double = .zero
+
+        /// How fast particles should spin.
+        /// Defaults to zero
+        public var angularSpeed: SIMD3<Double> = [0, 0, 0]
+
+        /// How much variation to allow in particle spin speed.
+        /// Defaults to zero
+        public var angularSpeedVariation: SIMD3<Double> = [0, 0, 0]
+
+        // These properties determine how particles are drawn.
+
+        /// How large particles should be drawn, where a value of 1, the default, means 100% of the image size.
+        public var size: Double = 1
+
+        /// How much variation to use for particle size.
+        /// Defaults to zero
+        public var sizeVariation: Double = .zero
+
+        /// How how much bigger or smaller this particle should be by the time it is removed.
+        /// This is used as a multiplier based on the particle's initial size, so if it starts at size
+        /// 0.5 and has a `sizeMultiplierAtDeath` of 0.5, the particle will finish
+        /// at size 0.25.
+        /// Defaults to zero
+        public var sizeMultiplierAtDeath: Double = .zero
+
+        /// How much to stretch this particle's image based on its movement speed. Larger values
+        /// cause more stretching.
+        /// Defaults to zero
+        public var stretchFactor: Double = .zero
+
+        /// What colors to use for particles made by this system. If `randomRamp` is used
+        /// then the VortexSystem initialiser will pick one possible color ramp to use.
+        /// A single, white, color is used by default.
+        public var colors: VortexSystem.ColorMode = .single(.white) 
+
+        /// VortexSettings initialisation.
+        /// - Parameters: None. Uses sensible default values on initialisation, with no parameters required.
+        public init() {}
+
+        /// Convenient init for VortexSettings initialisation.  Allows initialisation based on an existing settings struct, copies it into a new struct and modifiiesf it via a supplied closure
+        /// - Parameters:
+        ///  - from: `VortexSettings`
+        ///  The base settings struct to be used as a base. Defaullt settings will be used if not supplied.
+        ///  - : @escaping (inout VortexSettings)->Void
+        ///  An anonymous closure which will modify the settings supplied in the first parameter
+        /// e.g.
+        /// ```swift
+        /// let newFireSettings = VortexSystem.Settings(from: .fire ) { settings in
+        ///         settings.position = [ 0.5, 1.03]
+        ///         settings.shape = .box(width: 1.0, height: 0)
+        ///         settings.birthRate = 600
+        ///     }
+        /// ```
+        public init(
+            from base: VortexSystem.Settings = VortexSystem.Settings(),
+            _ modifiedBy: @escaping (_: inout VortexSystem.Settings) -> Void
```

Comment:
This is an odd shape for an initializer. What does it offer over code like this:

```swift
var settings = VortexSystem.Settings(from: .fire)
settings.idleDuration = 3
settings.angle = .whatever
```